applications/typechain-polkadot-follow-up-2.md

```diff
@@ -0,0 +1,221 @@
+# Typechain grant #3
+
+# [Typechain-Polkadot](https://github.com/727-Ventures/typechain-polkadot)
+
+- **Project Name:** [Typechain-Polkadot](https://github.com/727-Ventures/typechain-polkadot)
+- **Team Name:** [727.ventures](https://github.com/727-Ventures)
+- **Payment Address:** ERC 20 (DAI):
+0x352F779203202c99699CdA8cc95adF06CcC8abEf
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 3
+
+## Project Overview :page_facing_up:
+
+### Overview
+
+Nowadays, when technologies are growing faster and faster, we should think about optimizing different routine processes and making older stuff better. One of these optimizations is to make code typesafe that will be flexible in different situations.
+
+When a smart contract is being written, the front-end developer receives its file representation in the Application Binary Interface (ABI) format. One ABI per each contract, a new ABI for every contract update.
+
+Information about how to interact with a contract (methods names, arguments & returns types, etc.) is included in this ABI file. It is not relatively human-readable, so extracting that information becomes a challenge. We need to have correct type definitions for each contract in TypeScript.
+
+Interaction with blockchain is done with the polkadot.js library, which only has abstract definitions for the contract in use. Thus users' code cannot be typesafe. And Typechain-Polkadot can change it.
+
+This Grant is a follow-up to the previous grants, [first grant](https://github.com/w3f/Grants-Program/blob/master/applications/typechain-polkadot.md) and [second grant](https://github.com/w3f/Grants-Program/blob/master/applications/typechain-polkadot-follow-up.md).
+In this grant, we aim to integrate event subscription, allow users to create their plugins and expand the tool into more packages so the users can alo use different parts of the application alone if needed.
```

Comment:
```suggestion
In this grant, we aim to integrate event subscriptions, allow users to create their plugins, and expand the tool into more packages so the users can also use different parts of the application alone if needed.
```

Reply:
Can you elaborate on the plugin support? To me it's not clear what you mean by that. Maybe you could make a detailed example of a plugin use case that typechain would support after the scope of this grant is delivered.